ipv8/REST/trustchain_endpoint.py

```diff
@@ -65,8 +65,9 @@ def initialize(self, session):
             offset = int(request.query['offset'])
 
         return Response({
-            "blocks": [dict(block) for block in
-                       self.trustchain.persistence.get_recent_blocks(limit=limit, offset=offset)]
+            "blocks": [dict(block)
```

Comment:
This would simply append all blocks from all TrustChains in a single list right? If the same peer is re-used across different TrustChain communities, it is impossible for an application using this REST API to distinguish blocks created in the different communities by the same peer.

Maybe we should add the TrustChain community as a REST resource to the URL? It's not very pretty and convenient but at least, it allows us to distinguish blocks in different communities.

Reply:
I am torn on this decision.

Making it a REST resource would mean our out-of-the-box usage requires two calls instead of one (1st get the list of communities and find the right one and 2nd get the blocks). But it RESTful.

Alternatively, by making it an optional URL argument you would have out-of-the-box use for most use-cases that have a single Trustchain subclass loaded. But this is ugly and counter-REST.

Reply:
[Moving this back to WIP as either of these should be implemented]

Reply:
@egbertbouman had a suggestion (which I will follow): redirects. Meaning we will use the RESTful resource urls, but automatically redirect from `/trustchain/recent`.